cleanup inputs and use lib

GTrunSec opened the pull request:
the lib makes it simple and easy.

GTrunSec committed:
cleanup inputs and use lib

GTrunSec commented:
@danielphan2003  please review it. thx

GTrunSec committed:
add lib args

GTrunSec commented:
@blaggacao  how about rename the `vs-ext` to `devos-ext-lib` ?

GTrunSec committed:
spcified overlay name

danielphan2003 commented on vscode/overlay.nix:
```diff
@@ -0,0 +1,8 @@
+final: prev: {
+  vscode-utils = prev.vscode-utils // (prev.callPackage ./. { });
+  lib = prev.lib.extend (lfinal: lprev: { } // import ./. {
+    lib = final.lib;
+    vscode-utils = final.vscode-utils;
+    vscode = final.vscode;
```
```suggestion
    inherit (final) lib vscode-utils vscode;
```

GTrunSec replied:
thx

danielphan2003 commented:
Thanks for the changes, this has been way over due on my side.

danielphan2003 commented:
> @blaggacao how about rename the `vs-ext` to `devos-ext-lib` ? we can add the different attrs of overlay extending the pkgs.lib something like:
> 
> ```nix
> devos-ext-lib.overlay.vscode
> devos-ext-lib.overlay.vim
> ```


`devos-ext-lib` sounds good to me. Given the simplicity of this repo, `vim-ext` should be right at home with `vs-ext`.

blaggacao commented:
Merge this first? Then rename?

GTrunSec commented:
> Merge this first? Then rename?

yeh，can you do that?

GTrunSec committed:
nix-lint fix

run commmad: nix -Lv develop --no-write-lock-file github:hardenedlinux/nixpkgs-hardenedlinux -c nixpkgs-hardenedlinux nix-lint fix

blaggacao commented:
Renamed to `devos-ext-lib` :rocket:
